### ISSUE TITLE
MicronautAwsProxyRequest duplicates headers into MultiValueHeaders

### ISSUE THREAD

**Issue by choufraise** [micronaut-projects/micronaut-aws#1649]
https://github.com/micronaut-projects/micronaut-aws/blob/cb02ba1e3ea6fa07edd2dd314887bf556d433fce/function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/MicronautAwsProxyRequest.java#L551

Found in Micronaut 3.8.6 (`micronaut-function-aws-api-proxy:3.10.9`), but was working as expected in 3.8.5 (`micronaut-function-aws-api-proxy:3.10.5`).

When using AWS API Gateway Proxy, the `MicronautAwsProxyRequest` now duplicates headers when present in both MultiValueHeaders and SingleValueHeaders. A normal API Gateway Proxy Request looks like this:
```
{
  ...
  "multiValueHeaders": {
    ...
    "tenant": [
      "my-tenant"
    ],
    ...
  },
  "headers": {
    ...
    "tenant": "my-tenant",
    ...
  },
  ...
  "requestSource": "API_GATEWAY"
}
```
i.e. the headers are supplied in both `multiValueHeaders` and `headers` by API Gateway. This is expected. After this commit https://github.com/micronaut-projects/micronaut-aws/commit/57ac443e218b59563a50f1339ef12e903be98a94 the headers are now _appended_ into the `headers` as `headers.get(headerName).add(singleValueHeaders.get(headerName))`. 

When you then ask for the headers in `HttpRequest<?>` like `request.getHeaders().getAll("tenant")` you would in this case get a `List.of("my-tenant", "my-tenant")`, while before (as it _should_ be) you would get the single value like `List.of("my-tenant")`. 

Please fix this asap as it is not re-producable locally in Unit Tests using `@MicronautTest` and `HttpClient` and only appears when running live in AWS.

I wrote a very simple test case as proof:
```
class MicronautAwsProxyRequestTest {

  @Test
  void multiAndSingleValueHeadersShouldBeMerged() {
    // Given
    AwsProxyRequest awsProxyRequest = new AwsProxyRequest();
    awsProxyRequest.setHeaders(new SingleValueHeaders());
    awsProxyRequest.getHeaders().put("tenant", "my-tenant");
    awsProxyRequest.setMultiValueHeaders(new Headers());
    awsProxyRequest.getMultiValueHeaders().put("tenant", List.of("my-tenant"));
    SecurityContext securityContext = Mockito.mock(SecurityContext.class);
    Context lambdaContext = Mockito.mock(Context.class);
    ContainerConfig config = Mockito.mock(ContainerConfig.class);

    // When
    MicronautAwsProxyRequest<?> result = new MicronautAwsProxyRequest<>("/",
        awsProxyRequest, securityContext, lambdaContext, config);

    // Then
    assertThat(result.getHeaders().getAll("tenant")).isEqualTo(List.of("my-tenant"));
  }
}
```
The output in 3.8.6 (`micronaut-function-aws-api-proxy:3.10.9`) is:
```
expected: ["my-tenant"]
 but was: ["my-tenant", "my-tenant"]
``` 
The output in 3.8.5 (`micronaut-function-aws-api-proxy:3.10.5`) is Successful

**Comment by thobiaskarlsson** [comment]
Hi. No activity at all on this bug? I can't understand how this hasn't affected more people as the lib is acting very strangely after the issue was introduced. I thought Micronaut as an organisation would be more responsive. :O

**Comment by sdelamo** [comment]
checking this issue now

**Comment by sdelamo** [comment]
@thobiaskarlsson this issue is fixed in 3.10.10. Thanks for reporting

**Comment by thobiaskarlsson** [comment]
@sdelamo thank you very much! Sorry for my nagging, I'm currently swamped. Next time I find a bug I'll gladly contribute instead of rushing you. :smiley:
